.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,67 @@
+name: Integration Tests
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_protected == 'true' && github.sha || github.ref }}-{{ github.event_name }}
+  cancel-in-progress: true
+
+jobs:
+  integration-tests:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Set up Docker Compose
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y docker-compose
+
+      - name: Build Docker Image
+        run: |
+          docker-compose build
+
+      - name: Build and start services
+        run: |
+          docker-compose up -d
+
+      - name: Create GCP Credentials File
+        run: |
+          echo "$CREDS_TEST_HUBBLE" > ${{ runner.workspace }}/gcp-key.json
+        shell: bash
+        env:
+          CREDS_TEST_HUBBLE: ${{secrets.CREDS_TEST_HUBBLE}}
```

Comment:
No, since the secret is not logged anywhere. Even in the worst case secret is logged, github obsucures it with ***
https://stackoverflow.com/questions/59481933/how-can-i-extract-secrets-using-github-actions